### DISCUSSION TITLE
Multiple Model CRUD NestJS API recommended approach?

### DISCUSSION THREAD

**Discussion by mitsos1os** [prisma/prisma#7480] [Q&A] [answered]
Hi everyone.

I am currently in the process of evaluating Prisma for a project with NestJS.
In general, I find it quite useful and straight forward except for one basic pain point. This is related to using Generics and common service such as described in https://github.com/prisma/prisma/discussions/3929

So to make sure I am not missing anything, if we have let's say 5 DB Models in NestJS, named ModelA -> ModelE and each of them supporting a basic CRUD API such as:

- **GET** /api/model?filter=<FILTER_OBJECT>
-  **GET** /api/model/<ID>
- **POST** /api/model (body includes new instance data)
- **PATCH** /api/model/<ID> (body includes update data)
- **DELETE** /api/model/<ID>

What is the recommended approach of creating type safe endpoints-services in NestJS? Separate Service - Controller for each model?
Is there a way of grouping it such as in TypeORM where you could create a BaseService using Generics that would call the appropriate repository _(ex `prisma.modelA` in our case)_ for each method? ex `findMany`, `findUnique`, `update` etc..?

**Comment by ryands17** [comment, marked as answer]
@mitsos1os 👋 

Currently there's no easy way to dynamically fetch the model in Prisma and we will be looking into this so it would be great if you could add a 👍 to this request so that we can check the priority for this. https://github.com/prisma/prisma/issues/5273

As for dynamic models, you can do this but TypeScript will complain as it is not straightforward if you check the workarounds as well provided in the above issue. Currently you would need to add separate routes for the model or conditionally check for each model in that case.

**Reply by mitsos1os** [reply]
Thanks @ryands17 . I will move my comments to the feature request.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
